File: Backend/Models/Entities/Budget_Line.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Budget_Line
    {
        [Key]
        [Required]
        public int BudgetLineId { get; set; }

        [Required]
        public string Account_Code { get; set; }

        [Required]
        public int Budget_ID { get; set; }

        [Required]
        public int Category_ID { get; set; }

        [ForeignKey("Budget_ID")]
        public Budget_Allocation Budget_Allocation { get; set; }

        [ForeignKey("Category_ID")]
        public Budget_Category Budget_Category { get; set; }
        
        public string Month { get; set; }
        public decimal BudgetAmt { get; set; }
        public decimal ActualAmt { get; set; }
        public decimal Variance { get; set; }
    }
}

File: Frontend/src/app/Shared/BudgetLine.ts
import { BudgetCategory } from "./BudgetCategory";
import { BudgetAllocation } from "./BudgetAllocation";

export interface BudgetLine {
  budgetLineId: Number;
  budget_ID?: Number;
  category_ID: Number;
  account_Code: String;
  budget_Allocation?: BudgetAllocation;
  budget_Category: BudgetCategory;
  month: String;
  budgetAmt: Number;
  actualAmt: Number;
  variance: Number;
}
